glean-core/ffi/src/macros.rs

```diff
@@ -79,6 +81,29 @@ macro_rules! define_metric {
         }
         )?
 
+        $(
+        #[no_mangle]
+        pub extern "C" fn $test_get_num_recorded_errors_fn(
+            glean_handle: u64,
+            metric_id: u64,
+            error_type: i32,
+            storage_name: FfiStr
+        ) -> i32 {
+            crate::HandleMapExtension::call_infallible(&*crate::GLEAN, glean_handle, |glean| {
+                crate::HandleMapExtension::call_infallible(&*$metric_map, metric_id, |metric| {
+                    let error_type = std::convert::TryFrom::try_from(error_type).unwrap();
+                    let storage_name = crate::FallibleToString::to_string_fallible(&storage_name).unwrap();
+                    test_get_num_recorded_errors(
+                        glean,
+                        (metric as &dyn metrics::MetricType).meta(),
```

Comment:
No need for dynamic dispatch here.

```suggestion
                        glean_core::metrics::MetricType::meta(metric),
```